File: labs/lab10/frequencies.pl
#!/usr/bin/perl -w

# map letters to counts

my %letterCount;

while($line = <STDIN>) {
    chomp $line;

    # remove anything but letters and numbers
    $line =~ s/[^A-Za-z0-9]//g;

    # split the line into an array of characters
    @chars = split //, $line;
    foreach my $letter (@chars) {
        # record count in hash table
        $letterCount{$letter}++;
    }
}

# output count of each letter, sorted on the keys (letters)
foreach my $letter (sort keys %letterCount) {
    print "'$letter' occurred $letterCount{$letter} times\n";
}

File: labs/lab10/frequencies.py
#!/usr/bin/python

import fileinput, collections, sys, re

freq = collections.defaultdict(int)

for line in fileinput.input():
    chars = list(line)
    for c in chars:
        if c.isalnum():
            freq[c] += 1

for f in sorted(freq):
    # f-string is great
    print(f"'{f}' occurred {freq[f]} times")
